src/theory/builtin/abstract_type.h

```diff
@@ -0,0 +1,56 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * A class for representing abstract types.
+ */
+
+#include "cvc5_public.h"
+
+#ifndef CVC5__THEORY__BUILTIN__ABSTRACT_TYPE_H
+#define CVC5__THEORY__BUILTIN__ABSTRACT_TYPE_H
+
+#include "expr/kind.h"
+
+namespace cvc5::internal {
+
+/**
+ */
+class AbstractType
+{
+ public:
+  AbstractType(Kind k);
+  AbstractType(const AbstractType& op);
+
+  /** return the kind of the current object */
```

Comment:
Related to my comment below, what current object? Is this the kind of this abstract type? So, `ABSTRACT_TYPE`? Or the kind of the sort parameter?

Reply:
This is still TODO, right?